### PR TITLE
soc: arm: st_stm32: add support for STM32H7B0 SoC

### PR DESCRIPTION
**Pull request by CharlesDias**
Add the Kconfig configuration, define the maximum clock and add the DTS file to support the STM32H7B0 SoC.

### CODE REVIEW

File: soc/arm/st_stm32/stm32h7/Kconfig.soc [hunk 1]
```diff
@@ -83,6 +83,16 @@ config SOC_STM32H7B3XXQ
...
+config SOC_STM32H7B0XX
+	bool "STM32H7B0XX"
+	select CPU_CORTEX_M7
+	select CPU_HAS_FPU_DOUBLE_PRECISION
+
+config SOC_STM32H7B0XXQ
+	bool "STM32H7B0XXQ"
+	select CPU_CORTEX_M7
+	select CPU_HAS_FPU_DOUBLE_PRECISION
+
```
Comment (erwango) [inline soc/arm/st_stm32/stm32h7/Kconfig.soc:95]: Please use alphanumeric order.

File: dts/arm/st/h7/stm32h7b0.dtsi [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (c) 2023 Charles Dias <charlesdias.cd@outlook.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <mem.h>
+#include <st/h7/stm32h7b3.dtsi>
```
Comment (erwango) [inline dts/arm/st/h7/stm32h7b0.dtsi:8]: No, please get stm32h7b3.dtsi including stm32h7b0.dtsi.

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks for this contribution. Few changes required, otherwise looks good.

File: dts/arm/st/h7/stm32h7b0.dtsi (hunk 2 above)
Comment (CharlesDias) [reply dts/arm/st/h7/stm32h7b0.dtsi:8]: Sorry, I didn't get the suggestion.

Comment (erwango) [reply dts/arm/st/h7/stm32h7b0.dtsi:8]: I'd like to have `#include stm32h7b0.dtsi` in stm32h7b3.dtsi. Not the other way round

Comment (CharlesDias) [reply dts/arm/st/h7/stm32h7b0.dtsi]: Done. Thanks you!

File: soc/arm/st_stm32/stm32h7/Kconfig.soc (hunk 1 above)
Comment (CharlesDias) [reply soc/arm/st_stm32/stm32h7/Kconfig.soc]: Done. Thanks!

**Review by erwango — APPROVED** [review]
Thanks for the changes!

**Review by gautierg-st — APPROVED** [review]

**Pull request merged**